### ISSUE TITLE
/delete/{id} should not return 404 if item missing (e.g., already deleted)

### ISSUE THREAD

**Issue by shtrom** [wallabag/wallabag#2710]
:warning: If your issue is about an error during fetching a link, please read: http://doc.wallabag.org/en/master/user/errors_during_fetching.html#how-can-i-help-to-fix-that

### Issue details

See wallabag/android-app#375: when the deletion of an entry is queued for deletion in the app, but the entry has been delete (e.g., from the web interface) meanwhile, the app gets into an infinite battery-consuming loop trying to delete the no-longer existent item.

The root cause is that the `/delete/{id}` route returns a 404 (wrapped in a 200, for the `GET`) when trying to delete a non-existent entry. To better match with the semantics of the HTTP DELETE method, which should be idempotent in its response whether an object was deleted or not, this method should not return a 404, but a 200, whether an entry existed or not.

### Environment

* wallabag-2.1.5
* How did you install wallabag? OpenBSD ports https://scm.narf.ssji.net/git/openbsd-ports/tree/www/wallabag
* php version: 5.6
* OS: OpenBSD
* type of hosting (shared or dedicated): dedicated
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): SQLite

### Steps to reproduce/test case

1. Add entry
2. Sync in App
3. Delete entry in web interface
4. Delete entry in app
5. Sync in App

-> The app keeps getting a 404 trying to delete the already-deleted entry.

**Comment by nicosomb** [comment]
Hello @shtrom! 

Thanks for you post. 
I think we can debate for a long time on this subject. 

For example, why not a 204 status? 
404 status for a non existing entry is not an error I think. 

What about trying to delete an entry which has never exists? Should we return 200 too? Or 404? 

My opinion: 404 is a good choice. 

@j0k3r @tcitworld @bourvill any idea?

**Comment by j0k3r** [comment]
That's a long debate.
I'm ok with the 204.

Also, the Android app shouldn't retry endlessly an url which return 404. There might be a problem here too.

**Comment by bourvill** [comment]
Like j0k3r, i think the android app should doesn't retry endlessly on 404.

The status code it's a long debate, i don't know who is the best choice

**Comment by shtrom** [comment]
Yup, 204 makes sense. Up to you guys, as long as the app knows how to handle the reply, that's fine by me (:

**Comment by zaghadon** [comment]
Hey guys, I believe it's time to close the issue now. 

For your information, I had arrived here from an Open Source tree looking for open issues to fix 😉, it appears this wasn't an issue anyway, and had been fixed to the extent of opening expectations.
